src/modules/model.js

```diff
@@ -147,21 +155,54 @@ class Model {
     await this._callback('delete', id);
   }
 
-  async list({ indexName, indexMap, fields }) {
+  async list({
+    indexName,
+    indexMap,
+    fields,
+    limit = 20,
+    cursor = null
+  }) {
     // eslint-disable-next-line no-underscore-dangle
     this._before();
-    const iterator = this.mapper.query(this.MapperClass, indexMap, {
+    const projection = projectionUtil.format(fields);
+    const fieldsContainsId = projection.includes('id');
+    if (!fieldsContainsId) {
+      projection.push('id');
+    }
+
+    const {
+      lastEvaluatedKey = null,
+      scanIndexForward = true,
+      limit: queryLimit = limit,
+      currentPage = null
+    } = fromCursor(cursor);
+    const iterator = this.mapper.query(this.MapperClass, indexMap, Object.assign({
       indexName,
-      projection: objectFields.split(fields)
-    });
-    const resp = [];
+      projection,
+      scanIndexForward,
```

Comment:
I removed the comment I left this morning, I think it's a valid parameter for the list function